wear/src/main/java/io/homeassistant/companion/android/home/HomeActivity.kt

```diff
@@ -33,53 +60,28 @@ class HomeActivity : AppCompatActivity(), HomeView {
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
 
-        setContentView(R.layout.activity_home)
-
         DaggerPresenterComponent
             .builder()
             .appComponent((application as GraphComponentAccessor).appComponent)
             .presenterModule(PresenterModule(this))
             .build()
             .inject(this)
 
-        adapter = HomeListAdapter()
-        adapter.onSceneClicked = { entity -> presenter.onEntityClicked(entity) }
-        adapter.onButtonClicked = { id -> presenter.onButtonClicked(id) }
-
-        findViewById<WearableRecyclerView>(R.id.home_list)?.apply {
-            layoutManager = LinearLayoutManager(this@HomeActivity)
-            adapter = this@HomeActivity.adapter
+        if (presenter.onViewReady()) {
+            val entities = runBlocking {
```

Comment:
This should be a class variable, not just a function variable.  It should also use the mutable state:
`val entities by mutableStateListOf<Entity<Any>>()`

Reply:
Rather than running this blocking we should allow the UI to draw and load this on an IO thread then update the class variable of `entities`.  This will make it feel snappy on the first load.

Reply:
this is proving to be very difficult 🤔 would appreciate any hints :)

Reply:
ok looks like we had to switch to a view model for this, now recomposition triggers and its a lot faster, especially on subsequent reloads :)

Reply:
I ended up reverting changes to `onViewReady` as it was no longer necessary for recomposition